src/Control/Effect/Fail.hs

```diff
@@ -24,3 +25,9 @@ import qualified Control.Monad.Fail as Fail
 
 -- | @since 1.0.0.0
 type Fail = Throw String
+
+-- | @since 1.0.0.0
+pattern Fail :: String -> Fail m k
+pattern Fail s = Throw s
+
+{-# COMPLETE Fail #-}
```

Comment:
This is a handy convenience for constructing/matching `Fail` effects. Unfortunately, we can’t export it as part of the `Fail` export, because that trick only works for datatypes, and not type synonyms 😞